test/E2ETests/Smoke.E2E.Tests/SuiteLifeCycleTests.cs

```diff
@@ -70,74 +56,96 @@ private void ValidateTestRunLifecycle(string targetFramework)
             """);
 
         var caseClassCleanupWithCleanupBehaviorEndOfClass = RunEventsHandler.PassedTests.Single(x => x.TestCase.FullyQualifiedName.Contains("SuiteLifeCycleTestClass_ClassCleanupWithCleanupBehaviorEndOfClass"));
+        numberOfLines = caseClassCleanupWithCleanupBehaviorEndOfClass.Messages.Single().Text.Split('\n').Length;
+        Verify(numberOfLines == (targetFramework == "net6.0" ? 12 : 11)); // The number of the logs + 3 empety lines + 1 for the logs' header.
         caseClassCleanupWithCleanupBehaviorEndOfClass.Outcome.Should().Be(Microsoft.VisualStudio.TestPlatform.ObjectModel.TestOutcome.Passed);
         caseClassCleanupWithCleanupBehaviorEndOfClass.Messages.Single().Text.Should().Contain(
             targetFramework == "net6.0" ?
             """
+            ClassInitialize was called
             Ctor was called
             TestInitialize was called
             TestMethod was called
             TestCleanup was called
             DisposeAsync was called
             Dispose was called
+            ClassCleanup.EndOfClass was called
             """
             :
             """
+            ClassInitialize was called
             Ctor was called
             TestInitialize was called
             TestMethod was called
             TestCleanup was called
             Dispose was called
+            ClassCleanup.EndOfClass was called
             """);
 
         var caseClassInitializeWithInheritanceBehaviorBeforeEachDerivedClassAndClassCleanupWithInheritanceBehaviorNone = RunEventsHandler.PassedTests.Single(x => x.TestCase.FullyQualifiedName.Contains("SuiteLifeCycleTestClass_ClassInitializeWithInheritanceBehaviorBeforeEachDerivedClassAndClassCleanupWithInheritanceBehaviorNone"));
+        numberOfLines = caseClassInitializeWithInheritanceBehaviorBeforeEachDerivedClassAndClassCleanupWithInheritanceBehaviorNone.Messages.Single().Text.Split('\n').Length;
+        Verify(numberOfLines == (targetFramework == "net6.0" ? 11 : 10)); // The number of the logs + 3 empety lines + 1 for the logs' header.
         caseClassInitializeWithInheritanceBehaviorBeforeEachDerivedClassAndClassCleanupWithInheritanceBehaviorNone.Outcome.Should().Be(Microsoft.VisualStudio.TestPlatform.ObjectModel.TestOutcome.Passed);
         caseClassInitializeWithInheritanceBehaviorBeforeEachDerivedClassAndClassCleanupWithInheritanceBehaviorNone.Messages.Single().Text.Should().Contain(
             targetFramework == "net6.0" ?
             """
-            Ctor was called
-            TestInitialize was called
-            TestMethod was called
-            TestCleanup was called
-            DisposeAsync was called
-            Dispose was called
+            Base ClassInitialize.InheritanceBehaviorBeforeEachDerivedClass was called
```

Comment:
Having a full picture now, I think it'd be better to have a prefix property being string empty by default and can be overridden/set by children. So that when you are running project itself, there is no prefix (e.g. "Ctor was called") but when it's called in the context of a child class, the prefix "Base " is added.